### PR TITLE
Support IDs in GID format

### PR DESCRIPTION
**Pull request by swalkinshaw**
This allows for users to pass in an ID both integer and GID format.

### CODE REVIEW

File: test/node-query-test.js [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+  const config = new Config({
+    domain: 'sendmecats.myshopify.com',
+    storefrontAccessToken: 'abc123'
+  });
+
+  const client = new Client(config);
+
+  test('it converts the ID to GID format', () => {
+    const productQuery = nodeQuery('Product', ['title']);
+    const query = client.graphQLClient.query((root) => {
```
Comment (swalkinshaw) [inline test/node-query-test.js]: @jzjiang not sure if there's a more direct way to test this. Technically I can just pass a selection into `productQuery` but wasn't sure if this is the easiest way to do it.

File: src/node-query.js [hunk 2]
```diff
@@ -3,7 +3,16 @@ import createGid from './create-gid';
...
   return function(parentSelection, fieldName, id) {
-    parentSelection.add(fieldName, {args: {id: createGid(type, id)}}, (node) => {
+    const gidProtocol = 'gid:';
+    let gid;
+
+    if (String(id).substr(0, gidProtocol.length) === gidProtocol) {
+      gid = id;
+    } else {
+      gid = createGid(type, id);
+    }
```
Comment (minasmart) [inline src/node-query.js]: Maybe pull this into a `normailzeId` function to keep it encapsulated and portable?

Comment (swalkinshaw) [inline src/node-query.js]: Will do, makes tests easier as well 👍

**Comment by swalkinshaw** [comment]
Updated. Replaced `createGid` with new `normalizeId`. It didn't seem worth it to me to keep two separate ones still.

**Comment by swalkinshaw** [comment]
No longer needed with opaque IDs.

**Pull request closed without merging**